backend/data.go

```diff
@@ -44,23 +44,23 @@ type QueryDataRequest struct {
 // A slice of DataQuery makes up the Queries property of a QueryDataRequest.
 type DataQuery struct {
 	// RefID is the unique identifier of the query, set by the frontend call.
-	RefID string
+	RefID string `json:"refId"`
 
 	// QueryType is an optional identifier for the type of query.
 	// It can be used to distinguish different types of queries.
-	QueryType string
+	QueryType string `json:"queryType"`
 
 	// MaxDataPoints is the maximum number of datapoints that should be returned from a time series query.
-	MaxDataPoints int64
+	MaxDataPoints int64 `json:"maxDataPoints"`
 
 	// Interval is the suggested duration between time points in a time series query.
-	Interval time.Duration
+	Interval time.Duration `json:"intervalMs"`
 
 	// TimeRange is the Start and End of the query as sent by the frontend.
-	TimeRange TimeRange
+	TimeRange TimeRange `json:"timeRange"`
 
 	// JSON is the raw JSON query and includes the above properties as well as custom properties.
-	JSON json.RawMessage
+	JSON json.RawMessage `json:"json"`
```

Comment:
Since this is the object that is sent, I don't think this one would have a json key?

Reply:
good point -- since this property includes everything else it shoud likely be:

```suggestion
	JSON json.RawMessage `json:"-"`
```

Reply:
I don't think that's what you want @ryantxu. Have a look at https://github.com/grafana/grafana/blob/b3a868169be04b6935c813d9e8b3e1a5cc54735d/pkg/plugins/datasource/wrapper/datasource_plugin_wrapper_v2.go#L77-L82

If you want to encode DataQuery as JSON it should either have `JSON json.RawMessage `json:"json"`` or `JSON json.RawMessage` otherwise the JSON property will not be part of the encoded JSON.

Reply:
I confirm that as @marefr says if we change `json:"json"` to `json:"-"` the JSON property is not included.